modules/terraform/output.go

```diff
@@ -131,16 +135,71 @@ func OutputMapE(t *testing.T, options *Options, key string) (map[string]string,
 	return resultMap, nil
 }
 
-// OutputForKeysE calls terraform output for the given key list and returns values as a map.
+// OutputForKeys calls terraform output for the given key list and returns values as a map.
 // If keys not found in the output, fails the test
 func OutputForKeys(t *testing.T, options *Options, keys []string) map[string]interface{} {
 	out, err := OutputForKeysE(t, options, keys)
 	if err != nil {
 		t.Fatal(err)
 	}
+
 	return out
 }
 
+// OutputForKeysTgE calls terragrunt output-all for the given key list and returns values as a map.
+// If keys not found in the output, fails the test
+func OutputForKeysTgE(t *testing.T, options *Options, keys []string) (map[string]interface{}, error) {
+	out, err := RunTerraformCommandE(t, options, "output-all", "-no-color", "-json")
+	if err != nil {
+		return nil, err
+	}
+
+	// Parse out json from terragrunt output
```

Comment:
Note: I didn't get a chance to review this section because I don't quite have time to do a deep dive of this now, but promise to do that later today.